ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/validator/preexecution/PreExecutingValidatingCommitterSpec.scala

```diff
@@ -0,0 +1,150 @@
+
+class PreExecutingValidatingCommitterSpec
+    extends AsyncWordSpec
+    with Matchers
+    with MockitoSugar
+    with ArgumentMatchersSugar {
+
+  import PreExecutingValidatingCommitterSpec._
+  import PreExecutionTestHelper._
+
+  "commit" should {
+
+    "write the transactions when no post execution conflicts are detected" in {
+      val fixture = new ValidatingCommitterFixture
+      fixture.validatorReturnsSuccess
+      when(
+        fixture.conflictDetector.detectConflicts(
+          any[PreExecutionOutput[TestReadSet, TestWriteSet]],
+          any[StateReader[DamlStateKey, TestValue]],
+        )(any[ExecutionContext])
+      )
+        .thenReturn(Future.unit)
+      val submissionAggregator = mock[SubmissionAggregator]
+      when(fixture.ledgerDataExporter.addSubmission(any[SubmissionInfo]))
+        .thenReturn(submissionAggregator)
+      when(
+        fixture.postExecutionWriter.write(any[TestWriteSet], any[LedgerStateWriteOperations[Long]])(
+          any[ExecutionContext]
+        )
+      )
+        .thenReturn(Future.successful(SubmissionResult.Acknowledged))
+      fixture.committer
+        .commit(
+          aParticipantId,
+          "corid",
+          Raw.Envelope.empty,
+          Instant.now(),
+          new FakeStateAccess(mock[LedgerStateOperations[Unit]]),
+        )
+        .map(result => {
+          verify(fixture.postExecutionWriter)
+            .write(any[TestWriteSet], any[LedgerStateWriteOperations[Long]])(any[ExecutionContext])
+          verify(submissionAggregator).finish()
+          result shouldBe SubmissionResult.Acknowledged
+        })
+    }
+
+    "drop transaction when post execution conflicts are detected" in {
+      val fixture = new ValidatingCommitterFixture
+      fixture.validatorReturnsSuccess
+      when(
+        fixture.conflictDetector.detectConflicts(
+          any[PreExecutionOutput[TestReadSet, TestWriteSet]],
+          any[StateReader[DamlStateKey, TestValue]],
+        )(any[ExecutionContext])
+      )
+        .thenReturn(Future.failed(new ConflictDetectedException()))
+      val submissionAggregator = mock[SubmissionAggregator]
+      when(fixture.ledgerDataExporter.addSubmission(any[SubmissionInfo]))
+        .thenReturn(submissionAggregator)
+      fixture.committer
+        .commit(
+          aParticipantId,
+          "corid",
+          Raw.Envelope.empty,
+          Instant.now(),
+          new FakeStateAccess(mock[LedgerStateOperations[Unit]]),
+        )
+        .map(result => {
+          verify(fixture.postExecutionWriter, never)
+            .write(any[TestWriteSet], any[LedgerStateWriteOperations[Long]])(any[ExecutionContext])
+          verify(submissionAggregator).finish()
+          result shouldBe SubmissionResult.Acknowledged
+        })
+    }
+  }
+
+}
+
+object PreExecutingValidatingCommitterSpec {
+
+  import MockitoSugar._
+  import PreExecutionTestHelper._
+
+  class ValidatingCommitterFixture {
```

Comment:
Normally I'm not a fan of bags of mocks, but in this case I think it's necessary.